fw/cache.c

```diff
@@ -3145,11 +3181,16 @@ tfw_cache_start(void)
 	if (!(cache_cfg.cache || g_vhost->cache_purge))
 		return 0;
 
-	for_each_node_with_cpus(i) {
+	if (r = tfw_init_node_cpus())
+		goto close_db;
```

Comment:
As I see in `tfw_init_node_cpus()` allocation of `c_nodes` may fail, thus `c_nodes` will be NULL. Then you do
```
close_db:
	for_each_node_with_cpus(i)
		tdb_close(c_nodes[i].db);
```
that cause null ptr dereference.

Why we even try to close db if we didn't open any dbs so far?